script/Multicall3Delegatecall.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {Script} from "forge-std/Script.sol";
+
+/// @title Multicall3Delegatecall
+/// @notice Aggregate delegate calls, ensuring each returns success if required.
+///         This contract is based on Multicall3 deployed on many evm chains
+///         at 0xcA11bde05977b3631167028862bE2a173976CA11. It has been modified to
+///         1. Perform delegatecalls instead of calls.
+///         2. Utility getter functions have been removed.
+contract Multicall3Delegatecall {
+    struct Call3 {
+        address target;
+        bool allowFailure;
+        bytes callData;
+    }
+
+    struct Result {
+        bool success;
+        bytes returnData;
+    }
+
+    /// @notice Aggregate calls, ensuring each returns success if required
+    /// @param calls An array of Call3 structs
+    /// @return returnData An array of Result structs
+    function aggregate3(Call3[] calldata calls) public payable returns (Result[] memory returnData) {
```

Comment:
I compared this function with https://github.com/mds1/multicall/blob/main/src/Multicall3.sol#L98 and it looks like the only difference is 

```solidity
(result.success, result.returnData) = calli.target.delegatecall(calli.callData);
```

instead of 

```solidity
(result.success, result.returnData) = calli.target.call(calli.callData);
```

This makes sense to me.